base/compiler/abstractinterpretation.jl

```diff
@@ -606,6 +606,9 @@ function abstract_call_method(interp::AbstractInterpreter, method::Method, @nosp
     if edge === nothing
         edgecycle = edgelimited = true
     end
+    edge_effects = override_edge_effects(edge_effects, sv, method, :consistent)
+    edge_effects = override_edge_effects(edge_effects, sv, method, :effect_free)
+    edge_effects = override_edge_effects(edge_effects, sv, method, :nothrow)
```

Comment:
You need to be a bit careful here. Nothrow on the outer method does not necessarily imply nothrow on the inner call. There could be an internal throw/catch.

Reply:
ok, if I understand correctly, `:nothrow` effect setting annotated on the currently-analyzed frame (i.e. "outer method") should be used at each time we want to observe its `:nothrow` effect property for some inference/optimization, rather than used for overriding the edge effect here.